internal/server/iigointernal/utilities_test.go

```diff
@@ -0,0 +1,161 @@
+package iigointernal
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+)
+
+func TestWithdrawFromCommonPoolThrowsError(t *testing.T) {
+	fakeGameState := gamestate.GameState{CommonPool: 100}
+	// Withdraw more than we have in it
+	valueToWithdraw := 120
+	err := WithdrawFromCommonPool(valueToWithdraw, &fakeGameState)
+	if err == nil {
+		t.Errorf("We can withdraw more from the common pool than it actually has.")
+	}
+}
+
+func TestWithdrawFromCommonPoolDeductsValue(t *testing.T) {
+	fakeGameState := gamestate.GameState{CommonPool: 100}
+	valueToWithdraw := 60
+	_ = WithdrawFromCommonPool(valueToWithdraw, &fakeGameState)
+	unexpectedAmountRemaining := fakeGameState.CommonPool != 40
+	if unexpectedAmountRemaining == true {
+		t.Errorf("Not withdrawing resources from CommonPool correctly.")
+	}
```

Comment:
use a test table to exercise more common pool values and values to withdraw more programatically

Reply:
Agreed and added